src/Valit/Errors/ValitRuleError.cs

```diff
@@ -1,21 +1,25 @@
+using System;
+
 namespace Valit.Errors
 {
     internal class ValitRuleError
     {
-        public string Message  { get; }
+        private readonly Func<string> _message;
+        public string Message => _message();
         public int? ErrorCode { get; }
 
-        private ValitRuleError(string message)
+        private ValitRuleError(Func<string> message)
         {
-            Message = message;
+            _message = message;
         }
 
         private ValitRuleError(int errorCode)
         {
             ErrorCode = errorCode;
+            _message = () => null;
```

Comment:
I'm not quite sure whether we should return null or string.Empty as default value. @arekbal  ?

Reply:
I would go with ""... ;) Let it be string.Empty then.

Reply:
I have updated the code to return string.Empty :-)